src/lsp/cobol_preproc/compdir_grammar.mly

```diff
@@ -30,10 +30,11 @@
 %token NE              "<>"         [@keyword (* symbol *) "<>"]
 
 %token CDIR_DEFINE     [@keyword ">>DEFINE", "$DEFINE"]
-%token CDIR_ELIF       [@keyword ">>ELIF", ">>ELSE-IF"]
-%token CDIR_ELSE       [@keyword ">>ELSE"]
-%token CDIR_END_IF     [@keyword ">>END-IF"]
-%token CDIR_IF         [@keyword ">>IF"]
+%token CDIR_ELIF       [@keyword ">>ELIF", "$ELIF", ">>ELSE-IF", "$ELSE-IF"]
+%token CDIR_ELSE       [@keyword ">>ELSE", "$ELSE"]
+%token CDIR_END        [@keyword "$END"]       (* Note: no `>>END` equivalent *)
+%token CDIR_END_IF     [@keyword ">>END-IF", "$END-IF"]
+%token CDIR_IF         [@keyword ">>IF", "$IF"]
 %token CDIR_SET        [@keyword ">>SET", "$SET"]
 %token CDIR_SOURCE     [@keyword ">>SOURCE", "$SOURCE"]
```

Comment:
I _think_ you mean [`$SOURCEFORMAT`](https://www.microfocus.com/documentation/visual-cobol/vc90/DevHub/HRCDRHCDIR63.html). right? Note that this is additional complicated as MF allows optional hyphens, so you also will see `$SOURCE-FORMAT`.

Note for `>>SOURCE` (I don't know if this also applies to MF's version, I'd just assume that) this is "local only" - not sure if that's handled correctly:

>  If a `SOURCE FORMAT` directive is specified in library text, the specified format shall be in effect until another `SOURCE FORMAT` directive is encountered or the end of the library text is reached. When the processing of that library text is completed, the reference format shall revert to the reference format that was in effect for the `COPY` statement that resulted in processing of that library text.

Reply:
Yes I think I coded the handling of `>>SOURCE` via `COPY` in the way the standards describe it.

About `$SOURCEFORMAT`, I could not see it in GnuCOBOL's preprocessor code, or its tests. I could find `$SET SOURCEFORMAT` though, which should already be handled. (I've just made a little tweak to handle `SOURCE-FORMAT` as well).

Reply:
> About `$SOURCEFORMAT`, I could not see it in GnuCOBOL's preprocessor code, or its tests. I could find `$SET SOURCEFORMAT` though

That was a mistake on my side, everything but the ISO ones (that are in a different part in the MF docs and of course only partial supported) all directives would be either from the command line with `-C`, specified in a directive file,  or in the code as `$SET`. So all is fine here.